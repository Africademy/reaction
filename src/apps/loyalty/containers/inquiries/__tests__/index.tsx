import * as React from "react"
import * as ReactTestUtils from "react-dom/test-utils"
import * as renderer from "react-test-renderer"

import { Inquiries } from "../index"

const inquiryProps = {
  user: {
    artwork_inquiries_connection: {
      edges: [
        {
          node: {
            id: "andy-warhol-skull",
            impulse_conversation_id: "1",
            artwork: {
              image: {
                url: "http://path.to.cat.pics",
              },
              artists: [{
                name: "Percy Z",
                href: "/percy-z",
              }],
            },
          },
        },
        {
          node: {
            id: "andy-warhol-skullz",
            impulse_conversation_id: null,
            artwork: {
              image: {
                url: "http://path.to.cat.pics",
              },
              artists: [{
                name: "Percy Z Without Conversation",
                href: "/percy-z-without-conversation",
              }],
            },
          },
        },
      ],
    },
  },
} as any

describe("login", () => {
  it("renders the inquiries listing", () => {
    const inquiries = renderer.create(<Inquiries {...inquiryProps}/>)
    expect(inquiries).toMatchSnapshot()
  })

  it("renders the artist information", () => {
    const inquiries = ReactTestUtils.renderIntoDocument(<Inquiries {...inquiryProps}/>)
    const aTags = ReactTestUtils.scryRenderedDOMComponentsWithTag(inquiries, "a")
    const artistLink = aTags.find(tag => tag.href === "/percy-z" && tag.textContent === "Percy Z")
    expect(artistLink).toBeTruthy()
    const noConvoLink = aTags.find(tag => tag.href === "/percy-z-without-conversation")
    expect(noConvoLink).toBeUndefined()
  })
})
