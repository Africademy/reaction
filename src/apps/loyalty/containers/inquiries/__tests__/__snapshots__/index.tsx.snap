exports[`inquiries renders the inquiries listing 1`] = `
<div
  className="bXZydp">
  <div
    className="hiqXMD">
    <div
      className="dHhWkf">
      <div
        className="gQelpA"
        style={undefined}>
        
      </div>
    </div>
    <a
      className="iDuDUI"
      href="/">
      Back To Artsy
    </a>
  </div>
  <header
    className="dtwjRD">
    <div
      className="header-title ccrmRr">
      Please select all works you purchased
    </div>
    <p
      className="header-subtitle fLunyJ">
      We will confirm submitted purchases with the galleries in order to qualify you for the program membership.
    </p>
  </header>
  <div
    className="artworks">
    <div
      className="gGgzvo">
      <div
        className="jhQCSn">
        <div
          className="erlken">
          <div
            className="fXCsUe">
            <div
              className="iCSZmV"
              onClick={[Function]}
              size={250}>
              <div
                className="hmGDjd"
                size={250}>
                <img
                  className="eltJDW"
                  size={250}
                  src="http://path.to.cat.pics" />
                <div
                  className="overlay-container hovered">
                  <div
                    className="dtjypd">
                    <div
                      className="gNTyvH"
                      selected={false}>
                      <div
                        className="kEzGjr"
                        style={undefined}>
                        
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="tEGyX">
                <div>
                  <div>
                    <strong>
                      <a
                        className="jehOuL"
                        href="/percy-z">
                        Percy Z
                      </a>
                    </strong>
                  </div>
                  <a
                    className="jehOuL"
                    href={undefined}>
                    <em />
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <footer
    className="footer">
    <p
      className="fTgPgI">
      If you purchased any works not included
      <br />
       above, please list them.
    </p>
    <textarea
      className="NrqUR"
      onChange={[Function]}
      placeholder="Artwork, Artist, Gallery" />
    <button
      className="dLWLJv"
      onClick={[Function]}>
      <span>
        Submit purchases
      </span>
    </button>
  </footer>
</div>
`;
